src/Gui/PreferencePages/DlgSettingsGeneral.cpp

```diff
@@ -425,18 +425,44 @@ void DlgSettingsGeneral::loadThemes()
 {
     ui->themesCombobox->clear();
 
-    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/MainWindow");
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath(
+        "User parameter:BaseApp/Preferences/MainWindow");
 
     QString currentTheme = QString::fromLatin1(hGrp->GetASCII("Theme", "").c_str());
 
     Application::Instance->prefPackManager()->rescan();
     auto packs = Application::Instance->prefPackManager()->preferencePacks();
+    QString currentStyleSheet = QString::fromLatin1(hGrp->GetASCII("StyleSheet", "").c_str());
+    QFileInfo fi(currentStyleSheet);
+    currentStyleSheet = fi.baseName();
+    QString themeClassic = QString::fromStdString("classic");  // handle the upcoming name change

```

Comment:
`QLiteralString` rather than converting to a `std::string` and then to a QString.